src/hint_processor/builtin_hint_processor/secp/bigint_utils.rs

```diff
@@ -7,14 +7,55 @@ use crate::{
         hint_processor_definition::HintReference,
     },
     serde::deserialize_program::ApTracking,
-    types::{exec_scope::ExecutionScopes, relocatable::MaybeRelocatable},
+    types::{
+        exec_scope::ExecutionScopes,
+        relocatable::{MaybeRelocatable, Relocatable},
+    },
     vm::{
         errors::{hint_errors::HintError, vm_errors::VirtualMachineError},
         vm_core::VirtualMachine,
     },
 };
 use felt::Felt;
-use std::collections::HashMap;
+use std::{borrow::Cow, collections::HashMap};
+
+#[derive(Debug, PartialEq)]
+pub(crate) struct BigInt3<'a> {
+    pub d0: Cow<'a, Felt>,
+    pub d1: Cow<'a, Felt>,
+    pub d2: Cow<'a, Felt>,
+}
```

Comment:
Why `Cow<_>`?

Reply:
I wanted to use references when possible instead of cloning values, and using references wasn't possible